openslides/core/static/js/core/site.js

```diff
@@ -661,6 +661,86 @@ angular.module('OpenSlidesApp.core.site', [
     }
 ])
 
+/* This directive provides a csv import template with pa
+ * Papa parse is used to parse the csv file. Accepted attributes:
+ * * change:
+ *   Callback if file changes. The one parameter is csv passing the parsed file
+ * * config (optional):
+ *   - accept: String with extensions: default '.csv .txt'
+ *   - encodingOptions: List with encodings. Default ['UTF-8', 'ISO-8859-1']
+ *   - parseConfig: a dict passed to PapaParse
+ */
+.directive('csvImport', [
+    function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'static/templates/csv-import.html',
+            scope: {
+                change: '&',
+                config: '=?',
+            },
+            controller: function ($scope, $element, $attrs, $location) {
+                // set config if it is not given
+                if (!$scope.config) {
+                    $scope.config = {};
+                }
+                if (!$scope.config.parseConfig) {
+                    $scope.config.parseConfig = {};
+                }
+
+                $scope.inputElement = angular.element($element[0].querySelector('#csvFileSelector'));
+
+                // set accept and encoding
+                $scope.accept = $scope.config.accept || '.csv';
+                $scope.encodingOptions = $scope.config.encodingOptions || ['UTF-8'];
+                $scope.encoding = $scope.encodingOptions[0];
+
+                $scope.parse = function () {
+                    var inputElement = $scope.inputElement[0];
+                    if (!inputElement.files.length) {
+                        $scope.change({csv: undefined});
+                        return;
+                    }
+
+                    var parseConfig = $scope.config.parseConfig;
```

Comment:
Ah. This way it is not possible to override some of the following config values with custom stuff. Maybe you have to apply the config stuff from $scope.config after setup of default values.

Maybe lodash _.defaults helps (or something similar).